src/golang/lib/repos/sqlite/operator.go

```diff
@@ -282,6 +282,46 @@ func (*operatorReader) GetRelationBatch(
 	return relations, err
 }
 
+func (*operatorReader) GetByEngineIntegrationID(
+	ctx context.Context,
+	integrationID uuid.UUID,
+	DB database.Database,
+) ([]models.Operator, error) {
+	query := fmt.Sprintf(`
+		SELECT DISTINCT %s FROM
+		operator, workflow_dag, workflow_dag_edge
+		WHERE
+		workflow_dag_edge.workflow_dag_id = workflow_dag.id
+		AND (
+			workflow_dag_edge.from_id = operator.id
+			OR workflow_dag_edge.to_id = operator.id
+		)
+		AND (
+			(
+				json_extract(operator.spec, '$.engine_config') IS NULL
+				AND (
+					json_extract(workflow_dag.engine_config, '$.airflow_config.integration_id') = $1
+					OR json_extract(workflow_dag.engine_config, '$.k8s_config.integration_id') = $1
+					OR json_extract(workflow_dag.engine_config, '$.lambda_config.integration_id') = $1
+					OR json_extract(workflow_dag.engine_config, '$.databricks_config.integration_id') = $1
+				)
+			)
+			OR (
+				json_extract(operator.spec, '$.engine_config.airflow_config.integration_id') = $1
+					OR json_extract(operator.spec, '$.engine_config.k8s_config.integration_id') = $1
+					OR json_extract(operator.spec, '$.engine_config.lambda_config.integration_id') = $1
+					OR json_extract(operator.spec, '$.engine_config.databricks_config.integration_id') = $1
```

Comment:
This seems brittle because every time we add a new compute engine, we will need to update this query to include the new engine. Maybe the right thing to do is to have an explicit `engine_config.integration_id` field (can be null) if this field is generalizable to all compute engine types. If so let's create a task for this. 

For now, instead of hardcoding, can we programmatically generate each condition across all compute engines? Assume we have access to (need to get rid of conda):
```
	computeIntegrations := map[Service]bool{
		Airflow:    true,
		Lambda:     true,
		Conda:      true,
		Databricks: true,
		Kubernetes: true,
		Spark:      true,
	}
```
We can then loop this dict and generate the sql query conditions.

Reply:
I think programmatically generating the conditions makes sense! We can keep this set in `models` package so that everything are specified in the same place. I'd like to be conservative on adding a new shared field as it requires some data migration and the assumption may not hold in future (for example, it's already not used in conda now)

Reply:
Right. Then as you suggested let's just put this set in `models` and generate the query conditions.